From c984acfc97742c4d239b3190ebc7f1d6bb1fd16c Mon Sep 17 00:00:00 2001
From: andrei varvara <andrei.varvara@freescale.com>
Date: Mon, 21 May 2012 22:03:53 +0000
Subject: [PATCH 193/518] Fixed bug. SA mng was not initialized.

initialized sa mng in care of parent sa.
removed useless variables from dpa_ipsec_remove_sa

Signed-off-by: Andrei Varvara <andrei.varvara@freescale.com>
Acked-by: Mihai Serb <mihai.serb@freescale.com>
[Grabbed from the branch, LINUX_IR5.2.0, of
https://git.freescale.com/git-private/cgit.cgi/ppc/alu-b4860/linux.git.]
Signed-off-by: Tiejun Chen <tiejun.chen@windriver.com>
---
 drivers/staging/fsl_dpa_offload/dpa_ipsec.c |   11 +++++------
 1 files changed, 5 insertions(+), 6 deletions(-)

diff --git a/drivers/staging/fsl_dpa_offload/dpa_ipsec.c b/drivers/staging/fsl_dpa_offload/dpa_ipsec.c
index a03c83a..6eac78e 100644
--- a/drivers/staging/fsl_dpa_offload/dpa_ipsec.c
+++ b/drivers/staging/fsl_dpa_offload/dpa_ipsec.c
@@ -3192,10 +3192,14 @@ static int remove_inbound_sa(struct dpa_ipsec_sa *sa)
 static int remove_outbound_sa(struct dpa_ipsec_sa *sa)
 {
 	struct dpa_ipsec_sa *child_sa;
-	struct dpa_ipsec_sa_mng *sa_mng;
 	int err;
 
 	if (sa_is_parent(sa)) {
+		struct dpa_ipsec_sa_mng *sa_mng;
+
+		BUG_ON(sa->dpa_ipsec);
+		sa_mng = &sa->dpa_ipsec->sa_mng;
+
 		child_sa = sa->child_sa;
 
 		err = sa_rekeying_outbound(child_sa);
@@ -3289,8 +3293,6 @@ static int remove_outbound_sa(struct dpa_ipsec_sa *sa)
  */
 int dpa_ipsec_remove_sa(int sa_id)
 {
-	struct dpa_ipsec *dpa_ipsec;
-	struct dpa_ipsec_sa_mng *sa_mng;
 	struct dpa_ipsec_sa *sa, *child_sa = NULL;
 	int ret = 0;
 
@@ -3299,9 +3301,6 @@ int dpa_ipsec_remove_sa(int sa_id)
 		return -EPERM;
 	}
 
-	dpa_ipsec = gbl_dpa_ipsec;
-	sa_mng = &dpa_ipsec->sa_mng;
-
 	sa = get_sa_from_sa_id(sa_id);
 	if (!sa) {
 		xx_pr_err("Invalid SA handle for SA id %d\n", sa_id);
-- 
1.7.5.4

